pkg/k8shandler/service_test.go

```diff
@@ -0,0 +1,383 @@
+package k8shandler
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	loggingv1 "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func TestCreateOrUpdateServices(t *testing.T) {
+	isControlled := true
+
+	tests := []struct {
+		desc    string
+		cluster *loggingv1.Elasticsearch
+		objs    []runtime.Object
+		wantSvc map[string]*corev1.Service
+		wantErr bool
+	}{
+		{
+			desc: "create services",
+			cluster: &loggingv1.Elasticsearch{
```

Comment:
There looks like a bunch of duplication here that IMO could be improved by initializing the base struct and then copying and modifying it as needed for the various use-cases

Reply:
IMHO, this kind of duplication is super welcome in tests. Consider keeping the fixtures separate as a good practice that enables parallel testing.